tikv-client-common/src/kv/key.rs

```diff
@@ -101,6 +102,14 @@ impl Key {
             Bound::Excluded(self)
         }
     }
+
+    #[inline]
+    pub fn as_encoded(&self) -> Key {
```

Comment:
I think we should use the type system to differentiate between an encoded and unencoded key by adding an `EncodedKey` type